sdk/storage/azure-storage-blob/azure/storage/blob/_encryption.py

```diff
@@ -261,28 +261,30 @@ def read(self, size: int = -1) -> bytes:
                     # No more data to read
                     break
 
-                self.current = self._encrypt_region(data)
+                self.current = encrypt_data_v2(data, self.nonce_counter, self.content_encryption_key)
+                # IMPORTANT: Must increment the nonce each time.
+                self.nonce_counter += 1
 
         return result.getvalue()
 
-    def _encrypt_region(self, data: bytes) -> bytes:
-        """
-        Encrypt the given region of data using AES-GCM. The result
-        includes the data in the form: nonce + ciphertext + tag.
 
-        :param bytes data: The data to encrypt.
-        :return: The encrypted bytes.
-        :rtype: bytes
-        """
-        # Each region MUST use a different nonce
-        nonce = self.nonce_counter.to_bytes(_GCM_NONCE_LENGTH, 'big')
-        self.nonce_counter += 1
+def encrypt_data_v2(data: bytes, nonce: int, key: bytes) -> bytes:
```

Comment:
This whole file is private and not exported so it doesn't really matter. The reason I didn't put the underscore is because I pull this function into another file (encryption_async). That seems to be the pattern I used for this file is that if the function is used by another file, don't add an _.